### PR TITLE
Add lesson 14 materials

### PR DESCRIPTION
**Pull request by RasimKhusaenov**
### Summary

Added lesson 14 materials.

### How it works

image
image
image

### Test plan

List of steps to manually test introduced functionality:

- Go to http://localhost:3000/graphiql
- Make request using schema:
```graphql
mutation signUp {
  signUp(
    input: {
      email: "user+2@example.com"
      password: "123456"
    }
  ){
    currentUser {
      email
      firstName
      lastName
    }
    errors {
      path
      message
    }
  }
}
```
- Make request using schema:
```graphql
mutation signIn {
  signIn(
    input: {
      email: "user+2@example.com"
      password: "123456"
    }
  ){
    accessToken
    errors {
      path
      message
    }
  }
}
```

### Deploy notes

Notes regarding deployment the contained body of work.
These should note any db migrations, services, scripts, etc.

### Checklist:

- [x] I have performed a self-review of my own code
- [x] My pull request is shorter than 500 lines

### CODE REVIEW

File: app/interactors/users/authenticate.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
+module Users
+  class Authenticate
+    include Interactor
+
+    delegate :credentials, to: :context
+
+    def call
+      context.fail!(errors: errors) unless user_authenticated?
+
+      context.access_token = ::JWT.encode(payload, jwt_secret, "HS256")
```
Comment (RasimKhusaenov) [inline app/interactors/users/authenticate.rb:10]: Use method

File: app/interactors/users/save.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    include Interactor
+
+    delegate :user_params, to: :context
+
+    def call
+      context.user = user
+
+      context.fail!(errors: errors) unless user.update(user_params)
+
+      context.user = user.reload
```
Comment (MaximStadnikov) [inline app/interactors/users/save.rb:12]: this line is redundant

File: app/interactors/users/save.rb [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    end
+
+    private
+
+    def user
+      @user ||= context.user || User.new
+    end
+
+    def errors
+      [{ message: "Wrong email or password" }]
```
Comment (MaximStadnikov) [inline app/interactors/users/save.rb:22]: i think it's better to show real errors here:
```ruby
def errors
  user.errors.full_messages
end
```

**Review by MaximStadnikov — CHANGES_REQUESTED** [review]

File: app/interactors/users/save.rb (hunk 3 above)
Comment (MaximStadnikov) [reply app/interactors/users/save.rb:20]: i tried it myself and i got error about password and email

**Pull request merged**